pkl-core/src/test/kotlin/org/pkl/core/RepositoryHygiene.kt

```diff
@@ -12,4 +18,35 @@ class RepositoryHygiene {
   fun `no remaining language snippet test selection`() {
     assertThat(LanguageSnippetTestsEngine().selection).isEqualTo("")
   }
+
+  @Test
+  fun `no output files exists for language snippets without an input`() {
+    val input = snippetsFolder.resolve("input")
+    val inputs = input.walk().filter {
+      it.extension == "pkl"
+    }.map {
+      val path = input.relativize(it).toString()
+      inputRegex.replace(path, "$1$2")
+    }.collect(Collectors.toSet())
+
+    val output = snippetsFolder.resolve("output")
+    output.walk().filter { it.isRegularFile() }.forEach {
+      val out = output.relativize(it).toString()
+      checkOutputHasInput(inputs, out)
+    }
+  }
+
+  private fun checkOutputHasInput(inputs: Set<String>, output: String) {
+    val fileTocheck = outputRegex.replace(output, "$1.pkl")
+    assertThat(inputs).contains(fileTocheck)
+  }
+
+  private val snippetsFolder: Path by lazy {
+    FileTestUtils.rootProjectDir.resolve("pkl-core/src/test/files/LanguageSnippetTests")
+  }
+
+  companion object {
+    private val inputRegex = Regex("(.*)[.][^.]*([.]pkl)")
+    private val outputRegex = Regex("(.*)[.][^.]+$")
```

Comment:
[nit] this seems a little easier to understand

```suggestion
    private val inputRegex = Regex("(.*)\\.[^.]*(\\.pkl)")
    private val outputRegex = Regex("(.*)\\.[^.]+$")
```